app/data_sources/gitlab.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime
+
+import requests
+import concurrent.futures
+
+from typing import Dict, List
+
+from indexing_queue import IndexingQueue
+from data_source_api.basic_document import BasicDocument, DocumentType
+from data_source_api.base_data_source import BaseDataSource, ConfigField, HTMLInputType
+from data_source_api.exception import InvalidDataSourceConfig
+from pydantic import BaseModel
+
+GITLAB_BASE_URL = "https://gitlab.com/api/v4"
+PROJECTS_URL = f"{GITLAB_BASE_URL}/projects?membership=true"
+
+
+class GitlabConfig(BaseModel):
+    access_token: str
+
+
+class GitlabDataSource(BaseDataSource):
+
+    def _parse_issues(self, documents: [], project_id: str, project_url: str):
+        issues_url = f"{GITLAB_BASE_URL}/projects/{project_id}/issues"
+
+        issues_response = self._session.get(issues_url)
+        issues_response.raise_for_status()
+        issues_json = issues_response.json()
+
+        for issue in issues_json:
+            last_modified = datetime.strptime(issue["updated_at"], "%Y-%m-%dT%H:%M:%S.%fZ")
+            if last_modified < self._last_index_time:
+                continue
+
+            documents.append(BasicDocument(
```

Comment:
- [ ] instead of passing the documents as a parameter, return them from the function